src/main/java/com/birdie/kafka/connect/json/SchemaTransformer.java

```diff
@@ -98,8 +100,15 @@ SchemaAndValue transformJsonValue(Object obj, String key) {
         } else if (obj == null) {
             return null;
         }
+        
+        Schema.Type objSchemaType = Values.inferSchema(obj).type();
+        
+        if (convertNumbersToDouble && objSchemaType == Schema.Type.INT64) {
```

Comment:
Let's hide the actual type's logic in a method, so you can also check for the other `INT` values: `INT32`, `INT16` and `INT8`. Shouldn't we also convert `FLOAT32` to `FLOAT64`? I don't think that's compatible.

```suggestion
        if (convertNumbersToDouble && isNumberType(objSchemaType)) {
```

Reply:
Ah yes you're right, will check for all these values